contexts/tx.tsx

```diff
@@ -0,0 +1,122 @@
+import { createContext, ReactNode, useContext } from 'react'
+import { TxRaw } from 'cosmjs-types/cosmos/tx/v1beta1/tx'
+import { isDeliverTxSuccess } from '@cosmjs/stargate'
+import { coins } from '@cosmjs/stargate'
+import useToaster, { ToastPayload, ToastTypes } from 'hooks/toast'
+import { useMeshClient } from 'client'
+import { useWallet } from '@cosmos-kit/react'
+
+// Context to handle simple signingClient transactions
+export interface Msg {
+  typeUrl: string
+  value: any
+}
+
+export interface TxOptions {
+  gas?: number
+  toast?: {
+    title?: ToastPayload['title']
+    message?: ToastPayload['message']
+    type?: ToastTypes
+    actions?: JSX.Element
+  }
+}
+
+export interface TxContext {
+  tx: (
+    msgs: Msg[],
+    options: TxOptions,
+    callback?: (log: any) => void,
+  ) => Promise<void>
+}
+
+export const Tx = createContext<TxContext>({
+  tx: () => new Promise(() => {}),
+})
+
+export function TxProvider({ children }: { children: ReactNode }) {
+  const { currentWallet } = useWallet()
+  const { client } = useMeshClient()
+  const signingCosmWasmClient = client?.signingCosmWasmClient
+
+  const toaster = useToaster()
+
+  // Method to sign & broadcast transaction
+  const tx = async (
+    msgs: Msg[],
+    options: TxOptions,
+    callback?: (log: any) => void,
+  ) => {
+    // Gas config
+    const fee = {
+      amount: coins(0, 'uosmo'),
```

Comment:
This seems like bad overrides. I think keplr has better pre-chain defaults.
Why not remove this from the client, and just let keplr do it?

Reply:
We use fee: 'auto' to defer to keplr